### DISCUSSION TITLE
Localization Support

### DISCUSSION THREAD

**Discussion by callmejm** [rainbow-me/rainbowkit#1091] [General]
Hi, is there any way to set localization without custom connect button & custom modal?

**Comment by DanielSinclair** [comment]
@callmejm Not currently. We'd like to accomplish that very soon. What language are you hoping to see?

**Reply by callmejm** [reply]
zh-CN

**Reply by callmejm** [reply]
Hi, any update on this ? waiting for good news !

**Reply by elboletaire** [reply]
I'm here because I need to translate the modal to catalan and spanish. Sad there's no support yet.

**Reply by DanielSinclair** [reply]
@elboletaire Now available with [v1.1.0](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%401.1.0). We currently support es-419 for Latin American Spanish, but can certainly try to add more languages. Want to help us test out the translations for Catalan?

**Comment by vimercati-samir** [comment]
It would be great if you could also pass custom translations so even if a specific language is not supported yet by rainbow kit you can always translate it by yourself. Maybe some kinda of object structure where the key is fixed and the value is the custom translation that the developer provides, for example if I wanted to add a custom translation for the Italian language: 

{ 
	modal: { 
		"what_is_a_wallet": "Che cos'è un wallet?",
	} 
} 

That would be really helpful 🙏

**Reply by DanielSinclair** [reply]
We don't currently support an API to pass in custom translations, but this is certainly something we can work on if it's still valuable to you. We hope to turnaround new language support very quickly as well.

**Comment by Dazedi** [comment]
We'd love to use this in Japan, but would like some localization support. 
If needed be, I can try to make a proposal and write the feature when I've got some time.
I'm not a translator so can't really help much with the translation portion of it.

You could have language option to RainbowKitProvider as ISO language code for languages that are supported (translations provided by RainbowKit), with `overrides` that'll require a separate object to override translations. 

Could be something like below

```
type Translations = Record<string, string>;

type Language = 'en-US' | 'ja-JP' | ...;

type Localization = Translations | { language: Language, overrides?: Translations };

interface RainbowKitProviderProps = {
   ...already existing props,
   localization: Localization,
}
```

**Reply by DanielSinclair** [reply]
@Dazedi now available with [v1.1.0](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%401.1.0). Japanese support works out of the box with the `ja-JP` locale. Let us know if you spot any low-quality translations

**Reply by Dazedi** [reply]
@DanielSinclair sweet, thanks! Will take a look

**Comment by DanielSinclair** [comment]
This is now available with [v1.1.0](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%401.1.0)
